samples/bidirectional-chat/csharp/Function.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Extensions.SignalRService;
+using Microsoft.AspNetCore.SignalR;
+
+namespace bidirectional_chat
+{
+    public class SimpleChat : ServerlessHub
+    {
+        private const string Hub = "simplechat";
+        private const string NewMessageTarget = "newMessage";
+        private const string NewConnectionTarget = "newConnection";
+
+        [FunctionName("negotiate")]
+        public static SignalRConnectionInfo GetSignalRInfo(
+            [HttpTrigger(AuthorizationLevel.Anonymous)] HttpRequest req,
+            [SignalRConnectionInfo(HubName = Hub, UserId = "{headers.x-ms-signalr-user-id}")] SignalRConnectionInfo connectionInfo)
+        {
+            return connectionInfo;
+        }
+
+        [FunctionName(nameof(Connect))]
+        public async Task Connect([SignalRTrigger]InvocationContext invocationContext)
```

Comment:
Can we change the name Connect/Disconnect to OnConnected/OnDisconnected? The server SDK uses onConnectedAsync/onDisconnectedAsync

Reply:
But the `connect` is event is triggered before the client is really connected. It's in the handshake step which is different from SDK.

Reply:
When we verify the connection whether it comes from ASRS, if it is then pass, if not then drop, seems don't need customer to handle. 

`Connect` looks confused to CX. Can we handle it internally? If we want to support custom auth, we can use the DI feature (in custom auth PR). 

How about add a `Connected` event when the connection is really connected? and if we want to drop the connection, add an `Abort` event, which is similar to server SDK?

Reply:
Updated. Now `connect` and `disconnect` are renamed to `connected` and `disconnected`. Also, when `connected` invoked, the client has already connected.
May be there need to have a separate event `auth` or `handshake`, but it's not necessary now.